File: 20191115/phone_number_list.py
def solution(phone_book):
    answer = True
    
    for standard_num in phone_book:
        standard = standard_num
        standard_size = len(standard);
        for phone_num in phone_book:
            if standard_num == phone_num:
                continue
            if standard in phone_num[:standard_size]:
                return False
    return True

File: 20191115/phone_number_list.cpp
#include <string>
#include <vector>
#include <string>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    
    for(int i = 0; i<phone_book.size(); i++){
        string str_standard = phone_book[i];
        int num_size = str_standard.size();
        for(int j = 0; j<phone_book.size(); j++){
            if (i == j) continue;
            if (str_standard == phone_book[j].substr(0, num_size)) {
                return false;
            }
        }
    }
    return true;
}